crates/router/src/core/payment_methods/surcharge_decision_configs.rs

```diff
@@ -303,10 +309,22 @@ pub async fn perform_surcharge_decision_management_for_saved_cards(
         .change_context(ConfigError::InputConstructionError)?;
 
     for customer_payment_method in customer_payment_method_list.iter_mut() {
+        #[cfg(not(feature = "v2"))]
+        let payment_token = customer_payment_method.payment_token.clone();
+
+        #[cfg(feature = "v2")]
+        let payment_token = customer_payment_method
+            .payment_token
+            .clone()
+            .get_required_value("payment_token")
+            .change_context(ConfigError::InputConstructionError)?;
+
         backend_input.payment_method.payment_method = Some(customer_payment_method.payment_method);
         backend_input.payment_method.payment_method_type =
             customer_payment_method.payment_method_type;
-        backend_input.payment_method.card_network = customer_payment_method
+
+        #[cfg(not(feature = "v2"))]
+        let card_network = customer_payment_method
```

Comment:
Write separate function with cfg feature instead of in-lining  cfg feature for variables.